### PR TITLE
fix(backend): auth check middleware for specific media access

### PR DESCRIPTION
**Pull request by matmair**
This adds capabilities to ensure accessing /media/ and other protected paths is possible with MFA for tokens. It also:
- refactors the middleware functions for clarity
- adds some more unit testing for the middleware catch these edge cases
- re-enables testing of mfa flows / login

Closes https://github.com/inventree/inventree-app/issues/705
see https://github.com/inventree/InvenTree/issues/10754

### CODE REVIEW

File: src/backend/InvenTree/InvenTree/test_auth.py [hunk 1]
```diff
@@ -222,3 +225,22 @@ def test_registration_email(self):
...
+    def test_auth_request(self):
+        """Test the auth_request view."""
+        url = reverse('auth-check')
+
+        # Logged in user
+        self.get(url)
+
+        # Inactive user
+        # TODO @matmair - this part of auth_request is not triggering currently
+        # self.user.is_active = False
```
Comment (SchrodingersGat) [inline src/backend/InvenTree/InvenTree/test_auth.py:238]: Do we need to block merging until this test is completed?

**Comment by SchrodingersGat** [comment]
@matmair thanks for taking a look at this - your approach seems very clean, just waiting on a few extra unit tests I suppose?

File: src/backend/InvenTree/InvenTree/test_auth.py (hunk 1 above)
Comment (matmair) [reply src/backend/InvenTree/InvenTree/test_auth.py:238]: no, it is more an indicator that it is not possible to reach that part of the code in the view

https://github.com/inventree/InvenTree/blob/726e852b7be00be4f9e09c75da76cab43ab5ac65/src/backend/InvenTree/InvenTree/views.py#L15-L20

As this is a view, it is only reached after we already have passed AuthRequiredMiddleware - that will 401 if the user is not authed or inactive.

**Comment by matmair** [comment]
@SchrodingersGat ready for review and merge, should fix the MFA issues for app users

**Comment by SchrodingersGat** [comment]
@matmair thanks for the fixes :)

**Comment by matmair** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|1.1.x|[](https://github.com/inventree/InvenTree/pull/10795)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)
